### PR TITLE
Minor README polish

### LINKED ISSUE #37: Rework the README and branding
- [x] update tagline in gemspec and on GitHub: `The wise choice for Ruby memoization`
- [x] add feature list/unique selling points of our gem:
  - Highest performance + Rigorous benchmarking
  - Preset
  - reset
  - Frozen instances (value objects)
  - Test coverage
  - Well-documented API
- [x] Add other GitHub badges (see examples here: https://github.com/JacobEvelyn/friends)

### PR DESCRIPTION
**Pull request by JacobEvelyn**
**THIS IS NOT YET READY TO MERGE! SEE CHECKLIST IN #37**

Closes #37

### CODE REVIEW

**Review by ms-ati — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -94,18 +112,18 @@ git commits and tags, and push the `.gem` file to
...
 ## Documentation
 
-#### Documentation is Automatically Generated
+### Documentation is Automatically Generated
 
 We maintain API documentation using [YARD](https://yardoc.org/), which is
 published automatically at
-[RubyDoc.info](https://rubydoc.info/github/panorama-ed/memo_wise/main). To edit
-documentation locally and see it rendered in your browser, run:
+[RubyDoc.info](https://rubydoc.info/github/panorama-ed/memo_wise/MemoWise). To
```
Comment (JacobEvelyn) [inline README.md:119]: @ms-ati I'm interested in your thoughts here. My thinking is that the old link just takes people to what appears to be the README again, while this new link takes people to more useful documentation. Thoughts?

**Review by jemmaissroff — APPROVED** [review]
Looks like there's a small merge conflict, but otherwise lgtm!

File: README.md (hunk 1 above)
Comment (ms-ati) [reply README.md:122]: Great idea!

**Review by ms-ati — APPROVED** [review]
LGTM

**Pull request merged**